mirror-wordpress-plugins

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env ruby
+
+require "bundler/setup"
+require "dotenv"
+require "English"
+require "json"
+require "rubygems"
+require "zip"
+
+Dotenv.load
+
+# Generic error class, can be caught but should not be raised.
+class PluginUpdateError < StandardError
+end
+
+# Could not retrieve information about a given plugin.
+class WordPressPluginApiError < PluginUpdateError
+end
+
+# Could not download a .zip file for a given plugin.
+class WordPressPluginDownloadError < PluginUpdateError
+end
+
+# Could not perform Git action on repo. Only useful for interacting with remotes.
+class GitError < PluginUpdateError
+end
+
+# A Plugin is a GitHub repository that mirrors a plugin from wordpress.org.
+class Plugin
+  attr_reader :slug, :github_url, :default_branch, :full_path_to_clone, :wordpress_api_url, :wordpress_download_link,
+    :latest_github_version, :latest_wordpress_version
+
+  @@wordpress_api_url = "https://api.wordpress.org/plugins/info/1.0/"
+
+  def initialize(name, github_url, default_branch)
+    @slug = name
+    @github_url = github_url
+    @default_branch = default_branch
+    @full_path_to_clone = File.join(Dir.pwd, @slug)
+    @wordpress_api_url = "#{@@wordpress_api_url}#{@slug}.json"
```

Comment:
I think we might want to do `@slug.downcase` here, e.g. compare the output of

```
curl -s https://api.wordpress.org/plugins/info/1.0/wp-recaptcha
```

and

```
curl -s https://api.wordpress.org/plugins/info/1.0/wp-reCAPTCHA
```

Reply:
I'd assumed here that our repo names were slugs!